src/database-layer/db.ts

```diff
@@ -619,18 +630,37 @@ if (maybePg != null) {
 				}
 			}
 
-			public override async getTxLevelLock(namespaceKey: string, key: number) {
+			public override async getTxLevelLock(
+				namespaceKey: string,
+				key: number,
+				blocking: boolean = true,
+			) {
 				validateTransactionLockParameter(key, 'key');
 				const namespaceId = transactionLockNamespaceMap[namespaceKey];
 				if (namespaceId == null) {
 					throw new Error(
 						`Transaction lock namespace ${namespaceKey} not registered.`,
 					);
 				}
-				await this.executeSql(`SELECT pg_advisory_xact_lock($1, $2);`, [
-					namespaceId,
-					key,
-				]);
+				try {
+					if (blocking) {
+						const result = await this.executeSql(
+							`SELECT pg_advisory_xact_lock($1, $2);`,
+							[namespaceId, key],
+						);
+						return result.rows?.[0]?.['pg_advisory_xact_lock'] === '';
+					} else {
+						const result = await this.executeSql(
+							`SELECT pg_try_advisory_xact_lock($1, $2);`,
+							[namespaceId, key],
+						);
+						return result.rows?.[0]?.['pg_try_advisory_xact_lock'] === true;
```

Comment:
I'm sure most (all?) of these `?` can be removed:
```suggestion
						return result.rows[0].pg_try_advisory_xact_lock === true;
```

Reply:
Thx!